pkg/api/query/v1_test.go

```diff
@@ -1099,19 +1099,21 @@ func TestParseStoreMatchersParam(t *testing.T) {
 			}},
 		},
 	} {
-		api := QueryAPI{
-			gate: gate.NewKeeper(nil).NewGate(4),
-		}
-		v := url.Values{}
-		v.Set(storeMatcherParam, tc.storeMatchers)
-		r := &http.Request{PostForm: v}
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			api := QueryAPI{
+				gate: gate.NewKeeper(nil).NewGate(4),
+			}
+			v := url.Values{}
+			v.Set(StoreMatcherParam, tc.storeMatchers)
+			r := &http.Request{PostForm: v}
 
-		storeMatchers, err := api.parseStoreMatchersParam(r)
-		if !tc.fail {
-			testutil.Assert(t, reflect.DeepEqual(storeMatchers, tc.result), "case %v: expected %v to be equal to %v", i, storeMatchers, tc.result)
-		} else {
-			testutil.NotOk(t, err)
-		}
+			storeMatchers, err := api.parseStoreMatchersParam(r)
+			if !tc.fail {
+				testutil.Equals(t, tc.result, storeMatchers)
```

Comment:
Can we add `testutil.Ok(t, err)` here?

Reply:
I tried but I cannot do that. `testutil.Ok(t, err)` causes a panic because here `api.parseStoreMatchersParam(r)` return `*api.ApiError` type, which is always not nil.

Reply:
You are correct in that regard but what I am saying is that the goal here is to check that no error has happened.
```go
				testutil.Equals(t, (*baseAPI.ApiError)(nil), err)
```
Works for me. WDYT?